openssl/src/x509/extension.rs

```diff
@@ -428,6 +440,195 @@ impl AuthorityKeyIdentifier {
     }
 }
 
+#[cfg(ossl110)]
+pub struct SbgpAsIdentifier {
+    critical: bool,
+    asn: Vec<(u32, u32)>,
+}
+
+#[cfg(ossl110)]
+impl Default for SbgpAsIdentifier {
+    fn default() -> SbgpAsIdentifier {
+        SbgpAsIdentifier::new()
+    }
+}
+
+#[cfg(ossl110)]
+impl SbgpAsIdentifier {
+    /// Construct a new `SbgpAsIdentifier` extension.
+    pub fn new() -> SbgpAsIdentifier {
+        SbgpAsIdentifier {
+            critical: false,
+            asn: Vec::new(),
+        }
+    }
+
+    /// Sets the `critical` flag to `true`. The extension will be critical.
+    pub fn critical(&mut self) -> &mut SbgpAsIdentifier {
```

Comment:
In its newer version `fn critical(&mut self, value: bool)` is might be useful to have this configuration option for some legacy systems ? I personally cannot think of a use case, but just in case?